Test/Altinn.Broker.Tests/LegacyFileControllerTests.cs

```diff
@@ -269,4 +270,63 @@ public async Task GetFileOverview_FileDoesNotExist_FileNotFound()
         // Assert
         Assert.Equal(System.Net.HttpStatusCode.NotFound, getResponse.StatusCode);
     }
+
+    [Fact]
+    public async Task Download_DownloadFile_Success()
+    {
+        // Arrange
+        string onBehalfOfConsumer = FileInitializeExtTestFactory.BasicFile().Recipients[0];
+
+        // Arrange - initialize file
+        var initializeFileResponse = await _senderClient.PostAsJsonAsync("broker/api/v1/file", FileInitializeExtTestFactory.BasicFile());
+        Assert.True(initializeFileResponse.IsSuccessStatusCode, await initializeFileResponse.Content.ReadAsStringAsync());
+        var fileId = await initializeFileResponse.Content.ReadAsStringAsync();
+
+        // Arrange - upload file
+        var initializedFile = await _senderClient.GetFromJsonAsync<FileOverviewExt>($"broker/api/v1/file/{fileId}", _responseSerializerOptions);
+        Assert.NotNull(initializedFile);
+        var uploadedFileBytes = Encoding.UTF8.GetBytes("This is the contents of the uploaded file");
+        using (var content = new ByteArrayContent(uploadedFileBytes))
+        {
+            content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+            var uploadResponse = await _senderClient.PostAsync($"broker/api/v1/file/{fileId}/upload", content);
+            Assert.True(uploadResponse.IsSuccessStatusCode, await uploadResponse.Content.ReadAsStringAsync());
+        }
+
+        // Act
+        var downloadedFile = await _legacyClient.GetAsync($"broker/api/legacy/v1/file/{fileId}/download?onBehalfOfConsumer={onBehalfOfConsumer}");
+        var downloadedFileBytes = await downloadedFile.Content.ReadAsByteArrayAsync();
+
+        // Assert
+        Assert.Equal(uploadedFileBytes, downloadedFileBytes);
```

Comment:
Should also assert that downloadedFileBytes matches uploadedFileBytes (and/or that downloadedFileBytes.length > 0)

Reply:
Added length check on downloaded file bytes.
The existing assert already checks that each byte value matches